kernel/src/grant.rs

```diff
@@ -39,14 +39,14 @@ pub struct Allocator {
 }
 
 pub struct Owned<T: ?Sized> {
-    data: Unique<T>,
+    data: NonNull<T>,
     appid: AppId,
 }
 
 impl<T: ?Sized> Owned<T> {
     unsafe fn new(data: *mut T, appid: AppId) -> Owned<T> {
         Owned {
-            data: Unique::new_unchecked(data),
+            data: NonNull::new_unchecked(data),
```

Comment:
I know the proposed change is just a substitution, but while we are at it I think we should also remove the unsafety of this function. The `data` pointer must come from somewhere (either the kernel or some app), at which point we should be able to prove that it's indeed non-null. So I think this function should rather take a `data: NonNull<T>` as input - and constructing this `NonNull` should be moved up to where the null check is actually performed.

Same comment on `kernel/src/mem.rs`.

Reply:
I had a brief look and the null pointer check is done just a couple function calls up the stack, so it shouldn't be too much work to remove this unnecessary unsafety.